src/hyperlink/test/test_url.py

```diff
@@ -1116,6 +1138,31 @@ def test_autorooted(self):
         self.assertEqual(normal_absolute.rooted, True)
         self.assertEqual(attempt_unrooted_absolute.rooted, True)
 
+    def test_rooted_with_port_but_no_host(self):
+        # type: () -> None
+        """
+        URLs which include a netloc separator are inherently rooted, regardless
+        of whether they include one because they specify an explicit host or
```

Comment:
"which include a netloc" followed by "whether they include one" is confusing me. I think you mean "regardless of how they came to include one…"